platform/configurator/configurator-impl/src/main/java/org/codice/ddf/admin/configurator/impl/PropertyOperation.java

```diff
@@ -0,0 +1,203 @@
+ */
+public abstract class PropertyOperation
+        implements OperationBase, Operation<Void, Map<String, String>> {
+    /**
+     * Transactional handler for creating property files
+     */
+    private static class CreateHandler extends PropertyOperation {
+        private CreateHandler(Path configFile, Map<String, String> configs) {
+            super(configFile, configs, false);
+        }
+
+        @Override
+        public Void commit() throws ConfiguratorException {
+            Map<String, String> propertyMap = new HashMap<>();
+            propertyMap.putAll(configs);
+            saveProperties(propertyMap);
+
+            return null;
+        }
+
+        @Override
+        public Void rollback() throws ConfiguratorException {
+            boolean delete = configFile.delete();
+            if (!delete) {
+                LOGGER.debug("Problem deleting properties file {} for rollback", configFile);
+            }
+
+            return null;
+        }
+    }
+
+    /**
+     * Transactional handler for deleting property files
+     */
+    private static class DeleteHandler extends PropertyOperation {
+        private DeleteHandler(Path configFile) {
+            super(configFile, Collections.emptyMap(), true);
+        }
+
+        @Override
+        public Void commit() throws ConfiguratorException {
+            boolean delete = configFile.delete();
+            if (!delete) {
+                LOGGER.debug("Problem deleting properties file {} for rollback", configFile);
+            }
+
+            return null;
+        }
+    }
+
+    /**
+     * Transactional handler for updating property files
+     */
+    private static class UpdateHandler extends PropertyOperation {
+        private final boolean keepIgnored;
+
+        UpdateHandler(Path configFile, Map<String, String> configs, boolean keepIgnored) {
+            super(configFile, configs, true);
+
+            this.keepIgnored = keepIgnored;
+        }
+
+        @Override
+        public Void commit() throws ConfiguratorException {
+            Map<String, String> propertyMap = new HashMap<>();
+
+            if (keepIgnored) {
+                propertyMap.putAll(currentProperties);
+            }
+            propertyMap.putAll(configs);
+            saveProperties(propertyMap);
+
+            return null;
+        }
+    }
+
+    final File configFile;
+
+    protected final Map<String, String> configs;
+
+    final Map<String, String> currentProperties;
+
+    private PropertyOperation(Path configFile, Map<String, String> configs,
+            boolean loadCurrentProps) {
+        this.configFile = configFile.toFile();
+        this.configs = new HashMap<>(configs);
+
+        currentProperties = new HashMap<>();
+        if (loadCurrentProps) {
+            Properties result;
+            try {
+                Properties props = new Properties();
+                try (FileInputStream in = new FileInputStream(configFile.toFile())) {
```

Comment:
That's an interesting point. I'm not sure what I would make of a file that couldn't be closed correctly after being opened, so I'm not sure if it should be treated as an error or not here.